filechanges/opt/musicbox/read_ini.sh

```diff
@@ -11,281 +11,277 @@
-
-	case $1 in
-
-	    --clean | -c )
-		CLEAN_ENV=1
-	    ;;
-
-	    --booleans | -b )
-		shift
-		BOOLEANS=$1
-	    ;;
 
-	    --prefix | -p )
-		shift
-		VARNAME_PREFIX=$1
-	    ;;
+	# Sanitise BOOLEANS - interpret "0" as 0, anything else as 1
+	if [ "$BOOLEANS" != "0" ]
+	then
+		BOOLEANS=1
+	fi
 
-	    * )
-		if [ -z "$INI_FILE" ]
-		then
-		    INI_FILE=$1
-		else
-		    if [ -z "$INI_SECTION" ]
-		    then
-			INI_SECTION=$1
-		    fi
-		fi
-	    ;;
 
-	esac
+	# }}} END Options
 
-	shift
-    done
+	# }}} END Deal with command line args
 
-    if [ -z "$INI_FILE" ] && [ "${CLEAN_ENV}" = 0 ] ;then
-	echo -e "Usage: read_ini [-c] [-b 0| -b 1]] [-p PREFIX] FILE"\
-	    "[SECTION]\n  or   read_ini -c [-p PREFIX]" >&2
-	cleanup_bash
-	return 1
-    fi
-
-    if ! check_prefix ;then
-	cleanup_bash
-	return 1
-    fi
-
-    local INI_ALL_VARNAME="${VARNAME_PREFIX}__ALL_VARS"
-    local INI_ALL_SECTION="${VARNAME_PREFIX}__ALL_SECTIONS"
-    local INI_NUMSECTIONS_VARNAME="${VARNAME_PREFIX}__NUMSECTIONS"
-    if [ "${CLEAN_ENV}" = 1 ] ;then
-	eval unset "\$${INI_ALL_VARNAME}"
-    fi
-    unset ${INI_ALL_VARNAME}
-    unset ${INI_ALL_SECTION}
-    unset ${INI_NUMSECTIONS_VARNAME}
-
-    if [ -z "$INI_FILE" ] ;then
-	cleanup_bash
-	return 0
-    fi
-    
-    if ! check_ini_file ;then
-	cleanup_bash
-	return 1
-    fi
-
-    # Sanitise BOOLEANS - interpret "0" as 0, anything else as 1
-    if [ "$BOOLEANS" != "0" ]
-    then
-	BOOLEANS=1
-    fi
-
-
-    # }}} END Options
-
-    # }}} END Deal with command line args
-
-    local LINE_NUM=0
-    local SECTIONS_NUM=0
-    local SECTION=""
-    
-    # IFS is used in "read" and we want to switch it within the loop
-    local IFS=$' \t\n'
-    local IFS_OLD="${IFS}"
-    
-    # we need some optional shell behavior (shopt) but want to restore
-    # current settings before returning
-    local SWITCH_SHOPT=""
-    pollute_bash
-    
-    while read -r line || [ -n "$line" ]
-    do
+	local LINE_NUM=0
+	local SECTIONS_NUM=0
+	local SECTION=""
+	
+	# IFS is used in "read" and we want to switch it within the loop
+	local IFS=$' \t\n'
+	local IFS_OLD="${IFS}"
+	
+	# we need some optional shell behavior (shopt) but want to restore
+	# current settings before returning
+	local SWITCH_SHOPT=""
+	pollute_bash
+	
+	while read -r line || [ -n "$line" ]
+	do
 #echo line = "$line"
 
-	((LINE_NUM++))
+		((LINE_NUM++))
 
-	# Skip blank lines and comments
-	if [ -z "$line" -o "${line:0:1}" = ";" -o "${line:0:1}" = "#" ]
-	then
-	    continue
-	fi
+		# Skip blank lines and comments
+		if [ -z "$line" -o "${line:0:1}" = ";" -o "${line:0:1}" = "#" ]
+		then
+			continue
+		fi
 
-	# Section marker?
-	if [[ "${line}" =~ ^\[[a-zA-Z0-9_]{1,}\]$ ]]
-	then
+		# Section marker?
+		if [[ "${line}" =~ ^\[.*\]$ ]]
+		then
+			# Set SECTION var to name of section (strip [ and ] from section marker)
+			SECTION="${line#[}"
+			SECTION="${SECTION%]}"
+			read SECTION <<<"${SECTION}"
+			if ! [[ "${line}" =~ ^[[:print:]]+$  ]] ;then
```

Comment:
OK, I thought that's what you had changed but I must just have misread the
diff.
On 4 Mar 2015 06:35, "John Cass" <notifications@github.com> wrote:

> In filechanges/opt/musicbox/read_ini.sh
> <https://github.com/woutervanwijk/Pi-MusicBox/pull/234#discussion_r25754908>
> :
>
> >
> > -	# Section marker?
> > -	if [[ "${line}" =~ ^\[[a-zA-Z0-9_]{1,}\]$ ]]
> > -	then
> > +		# Section marker?
> > +		if [[ "${line}" =~ ^\[.*\]$ ]]
> > +		then
> > +			# Set SECTION var to name of section (strip [ and ] from section marker)
> > +			SECTION="${line#[}"
> > +			SECTION="${SECTION%]}"
> > +			read SECTION <<<"${SECTION}"
> > +			if ! [[ "${line}" =~ ^[[:print:]]+$  ]] ;then
>
> If we want to avoid empty section markers then we probably need to replace
> ^[[:print:]]+$ with something like ^\[.*\S.*\]+$. But I haven't tested
> this as I didn't want to fiddle with the original script.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/woutervanwijk/Pi-MusicBox/pull/234/files#r25754908>.
>